### ISSUE TITLE
HL-ZASM doesn't support octal numbers

### ISSUE THREAD

**Issue by suunrider** [wiremod/wire#1035]
```#pragma set OutputBinaryListing true
db 0377;``` outputs ```db 377```. Also it's not in implemented as escape sequence in a string (e.g. `db "\377"`)

**Comment by AbigailBuccaneer** [comment]
Do we actually want to support octal numbers? They're kind of horrible in every way.

**Comment by suunrider** [comment]
To support C strings in HL-ZASM there should be octal numbers

**Comment by Pharap** [comment]
Hexidecimal I can understand wanting, but octal is a bit more obscure.

What benefit would octal numbers provide?

Also the escape sequence in a string is technically a different matter.
It's not unrelated but different enough that there are different considerations involved.

**Comment by AbigailBuccaneer** [comment]
I would like to have some semblance of C compatibility, but I'm not sure how I feel about supporting the most horrible bits of C.

In particular, it would be nice to have the same string syntax as C, but octal escape sequences are worse and uglier than regular octal integer literals in C, because they don't require a leading `0`.

**Comment by AbigailBuccaneer** [comment]
@PhoenixBlack opinions?

**Comment by AbigailBuccaneer** [comment]
The more I think about it the more I dislike this.

I'm currently working on a C compiler for the ZVM anyway, so if C's what you really want then you'll get to have it. :stuck_out_tongue_winking_eye:

**Comment by suunrider** [comment]
C compiler for zvm? i love you

**Comment by deltecent** [comment]
Octal output is great for entering code into the front panel of an Altair 8800. I would love to have an option to generate a ".oct" file that looks like:

```
;
;  Kill the Bit game by Dean McDaniel, May 15, 1975
;
;  Object: Kill the rotating bit. If you miss the lit bit, another
;	   bit turns on leaving two bits to destroy. Quickly 
;	   toggle the switch, don't leave the switch in the up
;	   position. Before starting, make sure all the switches
;	   are in the down position.	   
;
	org	0
	lxi	h,0		;initialize counter
	mvi	d,080h		;set up initial display bit
	lxi	b,0eh		;higher value = faster
beg:	ldax	d		;display bit pattern on
	ldax	d		;...upper 8 address lights
	ldax	d
	ldax	d
	dad	b		;increment display counter
	jnc	beg
	in	0ffh		;input data from sense switches
	xra	d		;exclusive or with A
	rrc			;rotate display right one bit
	mov	d,a		;move data to display reg
	jmp	beg		;repeat sequence
	end
```

Would generate the following output file:

```
0000:   041 000 000 026 200 001 016 000 
0010:   032 032 032 032 011 322 010 000 
0020:   333 377 252 017 127 303 010 000 
```
